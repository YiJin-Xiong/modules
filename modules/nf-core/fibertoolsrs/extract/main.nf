process FIBERTOOLSRS_EXTRACT {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/fibertools-rs:0.6.2--h3b373d1_0':
        'biocontainers/fibertools-rs:0.7.1--h3b373d1_0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.bed"), emit: bed
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def outbed = params.ft_extract_type ? "--${params.ft_extract_type} ${prefix}.bed" : "--all ${prefix}.bed"

    """
    ft \\
        extract \\
        $args \\
        --threads $task.cpus \\
        $bam \\
        $outbed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fibertools-rs: \$(ft --version | sed -E 's/.* ([0-9]+\\.[0-9]+\\.[0-9]+).*/\\1/' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    echo $args
    
    touch ${prefix}.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fibertools-rs: \$(ft --version | sed -E 's/.* ([0-9]+\\.[0-9]+\\.[0-9]+).*/\\1/' )
    END_VERSIONS
    """
}
