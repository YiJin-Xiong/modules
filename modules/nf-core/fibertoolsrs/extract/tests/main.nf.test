nextflow_process {

    name "Test Process FIBERTOOLSRS_EXTRACT"
    script "../main.nf"
    process "FIBERTOOLSRS_EXTRACT"

    tag "modules"
    tag "modules_nfcore"
    tag "fibertoolsrs"
    tag "fibertoolsrs/predictm6a"
    tag "fibertoolsrs/extract"

    test("homo_sapiens - bam") {

        setup {
            run("FIBERTOOLSRS_PREDICTM6A") {
                script "../../predictm6a/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/alz.ccs.bam', checkIfExists: true),
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = FIBERTOOLSRS_PREDICTM6A.out.bam.collect{ meta, bam -> bam }.map{ bam -> [[ id: 'test_extract'], bam]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
