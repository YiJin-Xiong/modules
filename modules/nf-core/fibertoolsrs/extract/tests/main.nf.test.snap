{
    "homo_sapiens - bam - m6a": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2865e0e42a3ebcb9bd921dbb6ed08b3"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2865e0e42a3ebcb9bd921dbb6ed08b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-22T20:21:39.463175727"
    },
    "homo_sapiens - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.gz:md5,3bdde73d10c5369f509cfac1e59b5f68"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2865e0e42a3ebcb9bd921dbb6ed08b3"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.gz:md5,3bdde73d10c5369f509cfac1e59b5f68"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2865e0e42a3ebcb9bd921dbb6ed08b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-22T20:21:27.057663017"
    },
    "homo_sapiens - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2865e0e42a3ebcb9bd921dbb6ed08b3"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2865e0e42a3ebcb9bd921dbb6ed08b3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-22T20:22:09.352468888"
    }
}